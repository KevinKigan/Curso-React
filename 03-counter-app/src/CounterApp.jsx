import React from 'react';
import PropTypes from 'prop-types'
import {useState} from 'react'

export const CounterApp = ({value}) => {

    const[ counter, setCounter ] = useState(value);

    const addOne = () => {
        setCounter(counter + 1)
    }
    const substractOne = () => {
        setCounter(counter - 1)
    }
    const reset = () => {
        setCounter(value)
    }

  return (
    <>
        <h1>CounterApp</h1>
        <h2> { counter } </h2>
        <button aria-label='btn-add' onClick={addOne}>+1</button>
        <button aria-label='btn-substract' onClick={substractOne}>-1</button>
        <button aria-label='btn-reset' onClick={reset}>Reset</button>
    </>
  )
}


CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}