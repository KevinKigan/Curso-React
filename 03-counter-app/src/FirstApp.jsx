import React from 'react';
import PropTypes from 'prop-types'

const newMessage = {
    message: 'Hola Mundo',
    title: 'Kevin'
}

const getResult = () =>{
    return 4+4;
}

export const FirstApp = ({title, subtitle, number}) => { 
  return (
    <>
    <h1 data-testid="test-title">{title}</h1>
    {/* <h1>{getResult()}</h1> */}

    {/* <code>{JSON.stringify(newMessage)}</code> */}
    <p>{subtitle}</p>
    <p>{number+1}</p>
    </>
  )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    number: PropTypes.number.isRequired,

}

FirstApp.defaultProps = {
    subtitle: "No hay subtitulo" 
}