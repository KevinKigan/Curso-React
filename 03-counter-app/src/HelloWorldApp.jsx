export const HelloWorldApp = () => {
    return (<h1>hola mundo </h1>)
}

