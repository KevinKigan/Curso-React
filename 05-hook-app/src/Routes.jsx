import { Navigate, createBrowserRouter } from "react-router-dom";
import { AboutPage, HomePage, LoginPage, MainApp } from "./08-useContext";
// import { ErrorPage } from "./ErrorPage";
 
 
export const getRoutes = () => createBrowserRouter([
    {
        path: "/",
        element: <MainApp />,
        errorElement: <Navigate to="/about" replace  />,
        children: [
            {
                path: "/",
                element: <HomePage />,
            },
            {
                path: "login",
                element: <LoginPage />,
            },
            {
                path: "about",
                element: <AboutPage />,
            },
            
        ]
    }
]);