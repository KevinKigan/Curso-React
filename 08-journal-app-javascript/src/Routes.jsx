import { Navigate, createBrowserRouter } from "react-router-dom";
import { JournalApp } from "./JournalApp";


export const getRoutes = () => createBrowserRouter([
    {
        path: "/*",
        element: <JournalApp />,
        errorElement: <Navigate to="/" replace  />,
        children: []
    }
])