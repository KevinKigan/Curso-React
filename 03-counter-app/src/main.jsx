import React from 'react'
import ReactDOM from 'react-dom/client'
import { HelloWorldApp } from './HelloWorldApp'
import { FirstApp } from './FirstApp'
import './styles.scss';
import { CounterApp } from './CounterApp';





ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    {/* <FirstApp title="Hola soy Kevin" number={100} /> */}
    <CounterApp value={1}></CounterApp>
  </React.StrictMode>,
)
