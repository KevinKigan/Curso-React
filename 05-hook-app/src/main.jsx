import React from 'react'
import ReactDOM from 'react-dom/client'
import './assets/css/styles.css'
import { HooksApp } from './HooksApp'
import { CounterApp } from './01-useState/CounterApp'
import { CounterWithCustomHook } from './01-useState/CounterWithCustomHook'
import { SimpleForm } from './02-useEffect/SimpleForm'
import { SimpleFormWithCustomHook } from './02-useEffect/SimpleFormWithCustomHook'
import { MultipleCustomHooks } from './03-example/MultipleCustomHooks'
import { FocuesScreen } from './04-useRef/FocuesScreen'
import { Memorize } from './05-memos/Memorize'
import { MemoHook } from './05-memos/MemoHook'
import { CallbackHook } from './05-memos/CallbackHook'
import { Padre } from './06-tarea-memo/Padre'
import './07-useReducer/intro-reducer'
import { TodoApp } from './07-useReducer/TodoApp'
import { MainApp } from './08-useContext/MainApp'
import {
  createBrowserRouter,
  RouterProvider,
} from "react-router-dom";

import { getRoutes } from './Routes';
 
 
const router = getRoutes();
 
ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <RouterProvider router={ router } />
  </React.StrictMode>,
)

