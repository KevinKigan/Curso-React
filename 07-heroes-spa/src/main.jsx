import React from 'react'
import ReactDOM from 'react-dom/client'
import './styles.css'
import { RouterProvider } from 'react-router-dom'
import { getRoutes } from './router/Routes'
import { AppRouter } from './router/AppRouter'

const router = getRoutes();

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <RouterProvider router={ router } />
  </React.StrictMode>,
)
