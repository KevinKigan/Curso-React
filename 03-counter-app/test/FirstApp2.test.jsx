const { render, screen } = require("@testing-library/react")
import { FirstApp } from '../src/FirstApp'

describe('Pruebas FirstApp', () => {

  const title = 'Hola soy Kev';
  const subtitle = 'Hola soy un subtitulo'
  const number = 123;

  test('Test equals to snapshot', () => {
    const { container } = render(<FirstApp title={title} number={number} />)
    expect(container).toMatchSnapshot();
  })

  test('Debe mostrar el titulo en un h1', () => {
    render(<FirstApp title={title} number={number} />)
    expect(screen.getByText(title)).toBeTruthy()
    expect(screen.getByRole('heading', {level: 1}).innerHTML).toContain(title)

  })

  test('Debe mostrar el subtitulo enviado por props', () => {
    render(<FirstApp title={title} subtitle={subtitle} number={number} />)
    expect(screen.getAllByText(subtitle).length).toBe(1)
    // expect(screen.getByRole('paragraph').innerHTML).toContain(subtitle)

  })
})