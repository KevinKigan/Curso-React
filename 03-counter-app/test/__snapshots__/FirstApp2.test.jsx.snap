// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas FirstApp Test equals to snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola soy Kev
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    124
  </p>
</div>
`;
